src/main/java/com/parkhomovsky/bookstore/service/implementation/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.parkhomovsky.bookstore.service.implementation;
+
+import com.parkhomovsky.bookstore.dto.cart.ShoppingCartDto;
+import com.parkhomovsky.bookstore.dto.item.CartItemDto;
+import com.parkhomovsky.bookstore.exception.EntityNotFoundException;
+import com.parkhomovsky.bookstore.exception.UserNotAuthenticatedException;
+import com.parkhomovsky.bookstore.mapper.CartItemMapper;
+import com.parkhomovsky.bookstore.mapper.ShoppingCartMapper;
+import com.parkhomovsky.bookstore.model.CartItem;
+import com.parkhomovsky.bookstore.model.ShoppingCart;
+import com.parkhomovsky.bookstore.model.User;
+import com.parkhomovsky.bookstore.repository.cart.ShoppingCartRepository;
+import com.parkhomovsky.bookstore.repository.item.CartItemRepository;
+import com.parkhomovsky.bookstore.repository.user.UserRepository;
+import com.parkhomovsky.bookstore.service.ShoppingCartService;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getUserShoppingCart()
```

Comment:
from the method name, this method should only return shopping cart, but does also some population of shopping cart